pkg/k8s/exposer.go

```diff
@@ -52,22 +59,69 @@ func ExposeAsService(namespace, name *string, tunnelPort int, scheme string, raw
 	deployment := newDeployment(*namespace, *name, tunnelPort, image, ctrPorts)
 
 	service := newService(*namespace, *name, ports)
-	d, err := deploymentsClient.Create(context.Background(), deployment, v1.CreateOptions{
-		TypeMeta:     v1.TypeMeta{},
-		DryRun:       nil,
-		FieldManager: "",
-	})
-	if err != nil {
-		return err
+
+	var d *appsv1.Deployment
+	var err error
+	deploymentCreated := false
```

Comment:
when does this value change?

Reply:
my bad, its supposed to be updated when deployment is created, updated commit.